mmrazor/core/recorders/method_outputs_recorder.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import functools
+from types import MethodType
+from typing import List
+
+from mmcv.utils import import_modules_from_strings
+from torch import nn
+
+from ..builder import RECORDERS
+from .base_recorder import BaseRecorder
+
+
+@RECORDERS.register_module()
+class MethodOutputsRecorder(BaseRecorder):
+    """Recorder for intermediate results which are ``MethodType``'s outputs.
+
+    Note:
+        Different from ``FunctionType``, ``MethodType`` is the type of methods
+        of class instances.
+
+    Args:
+        sources (List(str)): The names of the methods whose output needs to be
+            recorded.
+        import_modules (List(str)): The modules which source methods belong to.
+
+    Examples:
+        >>> import copy
+        >>> import torch
+        >>> from mmcv.cnn import ConvModule
+        >>> from mmrazor.core import build_recorder
+
+        >>> # example recorder config
+        >>> recorder_cfg = dict(
+        >>>     type='MethodOutputs',
+        >>>     sources=['ConvModule.forward'],
+        >>>     import_modules=['mmcv.cnn'])
+
+        >>> recorder_cfg_ = copy.deepcopy(recorder_cfg)
+        >>> recorder_cfg_['type'] = recorder_cfg['type'] + 'Recorder'
+        >>> ctx = build_recorder(recorder_cfg_)
+        >>> ctx.initialize()
+
+        >>> conv = ConvModule(1,1,1)
+        >>> with ctx:
+        >>>     res = conv(torch.randn(1,1,1,1))
+
+        >>> ctx.data_buffer
+        >>> {'mmcv.cnn.ConvModule.forward': [tensor([[[[0.]]]])]}
+        >>> ctx.get_record_data('mmcv.cnn.ConvModule.forward')
+        >>> [tensor([[[[0.]]]])]
+        >>> ctx.get_record_data('mmcv.cnn.ConvModule.forward', data_index=0)
+        >>> tensor([[[[0.]]]])
+    """
+
+    def __init__(self, sources: List, import_modules: List):
+        super().__init__()
+        self.sources: List(str) = sources
+        self.import_modules: List(str) = import_modules
```

Comment:
Is it possible to avoid the manual imports inside this module and the multiple "eval" and "exec" function?